core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -343,6 +343,15 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     value
   }
 
+  /** Control our logLevel. This overrides any user-defined log settings.
+   * @param logLevel The desired log level as a string.
+   * Valid log levels include: ALL, DEBUG, ERROR, FATAL, INFO, OFF, TRACE, WARN
+   * Invalid log level defaults to DEBUG
```

Comment:
Hey Holden - can you have it be that if we see an invalid level we throw an exception? Otherwise it could be confusing if a user simply has a typo or something. Something like:

```
val validLevels = Seq("ALL", "DEBUG", "ERROR", "FATAL", "OFF", "TRACE", "WARN")
if (!validLevels.contains(logLevel.toUpperCase) {
  throw new InvalidArgumentException(
    s"Supplied level $logLevel did not match one of: ${validLevels.mkString(',')}")
}
```